velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -1278,37 +1419,43 @@ int32_t DateTimeFormatter::format(
         } break;
 
         case DateTimeFormatSpecifier::TIMEZONE: {
-          // TODO: implement short name time zone, need a map from full name to
-          // short name
+          VELOX_USER_CHECK_NOT_NULL(
+              timezone,
+              "The time zone cannot be formatted if it is not present.");
           if (token.pattern.minRepresentDigits <= 3) {
-            VELOX_UNSUPPORTED("short name time zone is not yet supported");
-          }
-          if (timezone == nullptr) {
-            VELOX_USER_FAIL("Timezone unknown");
+            const std::string& abbrev =
+                tz::locateZone(timezone->id())
```

Comment:
could we instead extend the TimeZone object to carry the timezone abbreviated name as well, so here we could just:

`timezone->shortName()`

(also, you don't need to tz::locateZone() if you already have the TimeZone*)